### PR TITLE
Milestone 2 components and redux store

### PR DESCRIPTION
**Pull request by lekegitrepo**
### Milestone 2 components and redux store

**_Implement:_**

  - **_The App component_**
  - **_The BooksList component_**
  - **_The Book component_**
  - **_The BooksForm component_**
  - **_The Redux actions and reducers_**
  - **_And initialize the Redux store_**

### CODE REVIEW

File: src/components/book.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+const Book = ({ book: { id, title, category } }) => (
+  <tr>
+    <td>{ id }</td>
+    <td>{ title }</td>
+    <td>{ category }</td>
+  </tr>
+);
+
+Book.propTypes = {
+  book: PropTypes.instanceOf(Object).isRequired,
```
Comment (Oghenebrume50) [inline src/components/book.js]: using PropTypes.shape here will allow you set a type check that is more accurate

File: src/containers/bookform.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    this.handleSelect = this.handleSelect.bind(this);
+  }
+
+  handleChange({ target }) {
+    this.setState({ title: target.value });
+  }
+
+  handleSelect({ target }) {
+    this.setState({ category: target.value });
+  }
```
Comment (Oghenebrume50) [inline src/containers/bookform.js:20]: These can both be condensed into one function if the name of the target element is used as the setState object key

File: src/containers/bookform.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+      'Sci-Fi',
+    ];
+
+    const { title, category } = this.state;
+    return (
+      <form>
+        <input type="text" name="title" value={title} onChange={this.handleChange} />
+        <select name="category" onChange={this.handleSelect} value={category}>
+          {CATEGORIES.map((cat, i) => (
+            <option key={i} value={cat}>
```
Comment (Oghenebrume50) [inline src/containers/bookform.js]: So keys should not be indexes, this warning should have been given by React, it is bad practice, it messes up the DOM if something is removed from the collection of object

**Review by Oghenebrume50 — CHANGES_REQUESTED** [review]
### CHANGES REQUIRED 

Hi @lekegitrepo good job  :+1:  this project looks elegant, the code setup is great, and I would just like to point out a few changes, please see below :slightly_smiling_face: 

**Please make changes and request for another review** :slightly_smiling_face: 

**RAPHAEL NORIODE** :mask:

File: src/containers/bookform.js (hunk 2 above)
Comment (lekegitrepo) [reply src/containers/bookform.js:20]: Using the target as the name of the setState object key makes the input and the dropdown function in a strange way.

Comment (bellom) [reply src/containers/bookform.js:20]: If I may ask, strange way how?

**Note**: you will still need to condensed the two functions into one when you get to the next milestone which is one of the hard `requirement`
we should have `handleChange.bind` and `handleSubmit.bind`

**Review by bellom — APPROVED** [review]
## Hi @lekegitrepo  :clap: 

**Great work making all the necessary corrections from the previous review.**

## Congratulation :+1:  
_Go ahead and merge to dev and don't forget to submit project completion form.
Good luck in the next milestone :slightly_smiling_face:_ 

**Best,**

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

File: src/containers/bookform.js (hunk 2 above)
Comment (lekegitrepo) [reply src/containers/bookform.js:20]: If I change the function to something like this:
```
handleInput({ target }) {
    this.setState({ target: target.value });
}
```
The input and the dropdown do not work. I tried to find out why but I couldn't.

Comment (bellom) [reply src/containers/bookform.js:20]: What if we use destructuring to get the target and set value. 🤔 
```
    handleChange(e) {
    const { target: { value, name } } = e;
    this.setState({
      [name]: value,
    });
  }
```

Comment (lekegitrepo) [reply src/containers/bookform.js:20]: Thanks so much for this.

**Pull request merged**